src/info/info_command.c

```diff
@@ -47,13 +47,13 @@ static void renderIndexDefinitions(RedisModule_Reply *reply, IndexSpec *sp) {
   if (num_prefixes) {
     RedisModule_ReplyKV_Array(reply, "prefixes");
     for (int i = 0; i < num_prefixes; ++i) {
-      REPLY_SIMPLE_SAFE(rule->prefixes[i]);
+      REPLY_SIMPLE_SAFE(HiddenUnicodeString_GetUnsafe(rule->prefixes[i], NULL));
     }
     RedisModule_Reply_ArrayEnd(reply);
   }
 
   if (rule->filter_exp_str) {
-    REPLY_KVSTR_SAFE("filter", rule->filter_exp_str);
+    REPLY_KVSTR_SAFE("filter", HiddenString_GetUnsafe(rule->filter_exp_str, NULL));
```

Comment:
I don't really understand what we did here.
We wrapped the expression with a hiddenString, but we return it as is always.
If we're not afraid to return this expression - let's not have it wrapped as a hidden string, and if we are - shouldn't it's exposure depend on the obfuscation config param?

Reply:
You are right, this is a bug.
Added the obfuscate boolean to the function, will use it to conditionally obfuscate the strings.